python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -0,0 +1,325 @@
+        elif self.config.scalingType == "leastSquare":
+            if self.statsControl is None:
+                self._setUpStatistics(star.stamp_im.mask)
+            starIm = star.stamp_im.clone()
+            # rotate
+            starIm = afwMath.rotateImageBy90(starIm, nb90Rots)
+            # "unnormalize" star
+            starIm *= star.annularFlux
+            # make sure both components use the star mask
+            XY = starIm.clone()
+            XY.image.array *= model.image.array
+            XX = starIm.clone()
+            XX.image.array = model.image.array**2
+            # compute scaling factor
+            XYSum = afwMath.makeStatistics(XY, self.statsFlag, self.statsControl).getValue()
+            XXSum = afwMath.makeStatistics(XX, self.statsFlag, self.statsControl).getValue()
+            scalingFactor = XYSum / XXSum if XXSum else 0
+        else:
+            raise AttributeError(f'Unknown scalingType "{self.config.scalingType}"; implemented options '
+                                 'are "annularFlux" and "leastSquare".')
+        if inPlace:
+            model.image *= scalingFactor
+        return scalingFactor
+
+    def run(self, inputExposure, inputBrightStarStamps, inputExtendedPsf, skyCorr=None, dataId=None):
+        """Iterate over all bright stars in an exposure to scale the extended
+        PSF model before subtracting bright stars.
+
+        Parameters
+        ----------
+        inputExposure : `afwImage.exposure.exposure.ExposureF`
+            The image from which bright stars should be subtracted.
+        inputBrightStarStamps :
+                `lsst.meas.algorithms.brightStarStamps.BrightStarStamps`
+            Set of stamps centered on each bright star to be subtracted,
+            produced by running
+            `lsst.pipe.tasks.processBrightStars.ProcessBrightStarsTask`.
+        inputExtendedPsf : `lsst.pipe.tasks.extended_psf.ExtendedPsf`
+            Extended PSF model, produced by
+            `lsst.pipe.tasks.extended_psf.MeasureExtendedPsfTask`.
+        skyCorr : `lsst.afw.math.backgroundList.BackgroundList` or ``None``,
+                  optional
+            Full focal plane sky correction, obtained by running
+            `lsst.pipe.drivers.skyCorrection.SkyCorrectionTask`.
+        dataId : `dict` or `lsst.daf.butler.DataCoordinate`
+            The dataId of the exposure (and detector) bright stars should be
+            subtracted from.
+
+        Returns
+        -------
+        subtractorExp : `afwImage.ExposureF`
+            An Exposure containing every bright star's profile; its image can
+            then be subtracted from the input exposure.
+        invImages : `list` [`afwImage.MaskedImageF`]
+            A list of small images ("stamps") containing the model, each scaled
+            to its corresponding input bright star.
+        """
+        inputExpBBox = inputExposure.getBBox()
+        if self.config.doApplySkyCorr:
```

Comment:
Do we have to check if `skyCorr` is not None here (default), or can `applySkyCorr` handle that case?